addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -0,0 +1,394 @@
+                self._setSlideStateAsDone();
+            }
+            if (self.slide.quiz){
+                self._bindQuizEvents()
+                self.answeredQuestions = [];
+                $('.submit-quiz').click(self._onSubmitQuiz.bind(self))
+            }
+        },
+        _generateQuizLetters: function(){
+            var letters = []
+            for(var i = 65; i < 91; i++){
+                letters.push(String.fromCharCode(i))
+            }
+            return letters;
+        },
+        _onPlayerReady: function(event){
+        },
+        _onPlayerStateChange: function(event){
+            var self = this;
+            var tid;
+            clearInterval(self.tid);
+            if (event.data == YT.PlayerState.PLAYING && !self.done) {
+                self.tid = setInterval(function(){
+                    if(event.target.getCurrentTime){
+                        var currentTime = event.target.getCurrentTime();
+                        var totalTime = event.target.getDuration();
+                        if(totalTime && currentTime > totalTime - 30){
+                            clearInterval(self.tid);
+                            if(!self.slide.hasQuiz && !self.slide.done){
+                                self.slide.done = true;
+                                self._setSlideStateAsDone();
+                            }
+                        }
+                    }
+                }, 1000)
+            }
+            if(event.data == YT.PlayerState.ENDED){
+                self.player = undefined;
+                self._goToNextSlide();
+            }
+        },
+        _bindListEvents: function(){
+            var self = this;
+            $('.slide-list-cell').each(function(){
+                $(this).click(self._onListCellClick.bind(self))
+            });
+        },
+        _bindQuizEvents: function(){
+            var self = this;
+            self.slide.quiz.submitted = false;
+            if(!self.slide.done){
+                $('.quiz-answer').each(function(){
+                    $(this).click(self._onAnswerClick.bind(self))
+                })
+            }
+            $('.next-slide').click(self._goToNextSlide.bind(self));
+        },
+        _onAnswerClick: function(ev){
+            var self = this;
+            var target = $(ev.currentTarget)
+            if((self.answeredQuestions.indexOf(target.attr('question_id'))) === -1){
+                var id = target.attr('id')
+                var question_id = target.attr('question_id')
+                $('.quiz-answer[question_id='+question_id+']').removeClass('quiz-good-answer');
+                $('#'+id+' input[type=radio]').prop('checked', true);
+                target.addClass('quiz-good-answer');
+            }
+        },
+        _onKeyDown: function(ev){
+            var self = this;
+            switch(ev.key){
+                case "ArrowRight":
+                self._goToNextSlide();
```

Comment:
Missing one level of indentation on this line and the break;

Reply:
Also, I know that pressing left/right arrow on YT forwards the video by 5 seconds.
It may collide with this? Or maybe not since the event is caught by the iframe?